### ISSUE TITLE
weird behavior from onAuthStateChanged when porting from firebase 8 to firebase beta 9

### ISSUE THREAD

**Issue by Mkassabov** [firebase/firebase-js-sdk#4991]
### [REQUIRED] Describe your environment

  * Operating System version: Microsoft Windows 10 Pro 10.0.19042 Build 19042
  * Browser version: Chrome Version 93.0.4535.0 (Official Build) canary (64-bit)
  * Firebase SDK version: 9.0.0-beta.2
  * Firebase Product: auth

#### Relevant Code:

Was trying to port the following code from firebase 8 to firebase beta 9:

firebase 8: 
```javascript
export function authState() {
    return new Observable(function (subscriber) {
        const unsubscribe = auth.onAuthStateChanged(subscriber);
        return { unsubscribe: unsubscribe };
    });
}
```

firebase beta 9
```javascript
export function authState() {
    return new Observable(function (subscriber) {
        const unsubscribe = onAuthStateChanged(auth, subscriber);
        return { unsubscribe: unsubscribe };
    });
}
```

From my understanding of the firebase 8 and firebase beta 9 docs these should be equivalent. However the firebase beta 9 version throws 
```
(node:38636) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'isStopped' of undefined
    at Subscriber.next (node_modules\rxjs\dist\cjs\internal\Subscriber.js:67:18)
    at node_modules\@firebase\auth\dist\node\register-2a5ac87a.js:2680:43
```

**Note:** using rxjs 7.0.0-beta.14 in both situations

**Comment by Mkassabov** [comment]
after logging the auth object it seems that it has an error being generated but never thrown/logged:
```javascript
  _errorFactory: ErrorFactory {
    service: 'auth',
    serviceName: 'Firebase',
    errors: {
      'dependent-sdk-initialized-before-auth': 'Another Firebase SDK was initialized and is trying to use Auth before Auth is initialized. Please be sure to call `initializeAuth` or `getAuth` before starting any other Firebase SDK.'
    }
  },
```

for reference here is the entire file:
```javascript
import { initializeApp, getApps } from "firebase/app";
import { getAuth, onAuthStateChanged } from "firebase/auth";
import { Observable } from "rxjs";

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  //...config
};
// Initialize Firebase

let firebaseApp;
let firebaseAppArray = getApps();
if (!firebaseAppArray.length) {
  firebaseApp = initializeApp(firebaseConfig);
} else {
  firebaseApp = firebaseAppArray[0];
}

export const auth = getAuth(firebaseApp);

export function authState() {
  return new Observable(
    function (subscriber) {
      const unsubscribe = onAuthStateChanged(auth, subscriber);
      return { unsubscribe: unsubscribe };
    }
  );
}
```

I've tried moving `getAuth()` to be in the same place as `initializeApp` though that doesn't seem to help. I've also tried calling `getAuth()` without a parameter since that seems to be show in the docs in a few places but that just throws an error about undefined not having a `.container` property

**Comment by looptheloop88** [comment]
Hi @Mkassabov, thanks for the report. I think this is related and duplicate of #4937.
